broker-daemon/block-order-worker/index.js

```diff
@@ -310,28 +310,33 @@ class BlockOrderWorker extends EventEmitter {
    * @returns {Object} contains activeOutboundAmount and activeInboundAmount of orders/fills
    */
   async calculateActiveFunds (marketName, side) {
-    const blockOrders = await this.getBlockOrders(marketName)
-    const blockOrdersForSide = blockOrders.filter(blockOrder => blockOrder.side === side)
-
-    const fullBlockOrders = await Promise.all(
-      blockOrdersForSide.map(async (bo) => {
-        await Promise.all([
-          bo.populateOrders(this.ordersStore),
-          bo.populateFills(this.fillsStore)
-        ])
-        return bo
-      })
-    )
-
-    let activeOutboundAmount = Big(0)
-    let activeInboundAmount = Big(0)
+    const blockOrders = await this.getBlockOrders(marketName, { side })
 
-    for (let blockOrder of fullBlockOrders) {
-      activeOutboundAmount = activeOutboundAmount.plus(blockOrder.activeOutboundAmount())
-      activeInboundAmount = activeInboundAmount.plus(blockOrder.activeInboundAmount())
+    // If we have no block orders that are available for our particular side, then
+    // we assume no active orders are available and can simply return 0
+    if (!blockOrders.length) {
+      return { activeOutboundAmount: Big(0), activeInboundAmount: Big(0) }
     }
 
-    return { activeOutboundAmount, activeInboundAmount }
+    // We grab the firstId and lastId to create a blockorder range so that we can
+    // grab all fills and orders to calculate active funds
+    const firstId = blockOrders[blockOrders.length - 1].id
+    const lastId = blockOrders[0].id
+
+    const orders = await BlockOrder.getOrdersForRange(this.ordersStore, firstId, lastId)
```

Comment:
Couldn't this include orders that belong to block orders that have their ID in the range, but are not actually active? If we are filtering out the inactive ones anyway, why even filter by the block order id range? Is it a performance optimization?

Reply:
Yeah, for the performance optimization.

**Original Method:**
1. Grab all block orders
2. For EACH block order, populate orders and fills
3. For EACH block order, Filter for active orders and active fills (happened in the model)
4. For EACH block order, get the active inbound balance and active outbound balance

**New Approach**
1. Grab all block orders
2. Grab all orders/fills for a range of block orders
3. Filter all of the orders/fills to be active
4. Grab the active amounts from each group (order or fill)

The thing that was really killing us was these queries separately on each block order

Reply:
Couldn't we just skip to step 2 in the new approach and just grab all orders and fills and filter by active? Are we expecting significant benefit from filtering by block orders first?

Reply:
Yeah, at this point in the broker we can, if we have another market, then we will have to add an index or change how this section of the code works.

You make a good point here, one that I was trying to think through. By grabbing all of the block orders, we had an assumption that "This range of block orders is only for this type of market", but the current implementation does not take that into account